packages/liquidator/src/walletBalanceAlarm.js

```diff
@@ -0,0 +1,157 @@
+      });
+
+      throw error;
+    }
+  }
+
+  async calculateTargetLiquidatorSynthBalance() {
+    this.numOfOpenPositions = await this.financialContractClient.getAllPositions().length;
+    return this.minSponsorTokens * this.numOfOpenPositions;
+  }
+
+  async calculateTargetLiquidatorCollateralBalance() {
+    const ooReward = await this.financialContract.methods.ooReward().call();
+    this.numOfOpenPositions = await this.financialContractClient.getAllPositions().length;
+    return this.numOfOpenPositions * ooReward;
+  }
+
+  async checkDisputerBotBalanceAgainstStrategy(currentCollateralBalance) {
+    try {
+      // Disputer Bot wallet USDC collateral balance should be >= UsdcTargetWalletBalance
+      const targetUsdcWalletBalance = await this.calculateUsdcTargetBalance();
+      console.log("checkDisputerBotBalanceAgainstStrategy called...");
+      console.log("Current Collateral Balance: ", currentCollateralBalance);
+      console.log("Target USDC Wallet Balance: ", targetUsdcWalletBalance);
+
+      if (currentCollateralBalance < targetUsdcWalletBalance) {
+        this.logger.warn({
+          at: "Disputer#WalletBalanceAlarm",
+          message: `Disputer bot wallet balance is ${currentCollateralBalance.toString()} which is below the target USDC wallet balance threshold of ${targetUsdcWalletBalance.toString()}. Replenish disputer bot wallet balance immediately`,
+          numOfOpenPositions: `${this.numOfOpenPositions}`,
+          totalCollateralAmount: `${this.totalCollateralAmount}`,
+          disputeBondPercentage: `${this.disputerBondPercentage}`,
+          currentCollateralBalance: `${currentCollateralBalance}`,
+          targetUsdcWalletBalance: `${targetUsdcWalletBalance}`,
+        });
+        return true;
+      } else {
+        this.logger.info({
+          at: "Disputer#WalletBalanceAlarm",
+          message: `Current disputer bot wallet balance of ${currentCollateralBalance.toString()} meets the target USDC wallet balance threshold of ${targetUsdcWalletBalance.toString()}. Disputer bot wallet USDC balance is within the healthy range.`,
+          numOfOpenPositions: `${this.numOfOpenPositions}`,
+          totalCollateralAmount: `${this.totalCollateralAmount}`,
+          disputeBondPercentage: `${this.disputerBondPercentage / 1e18}`,
+          currentCollateralBalance: `${currentCollateralBalance}`,
+          targetWalletBalance: `${targetUsdcWalletBalance}`,
+        });
+        return false;
+      }
+    } catch (error) {
+      this.logger.error({
+        at: "Disputer#WalletBalanceAlarm",
+        message: "An error occurred during the calculation of the disputer bot wallet balance",
+        error: error,
+        stack: error.stack,
+      });
+
+      throw error;
+    }
+  }
+
+  async calculateUsdcTargetBalance() {
+    const allPositions = await this.financialContractClient.getAllPositions();
+    this.totalCollateralAmount = await allPositions.reduce((total, position) => {
```

Comment:
I think we can directly use `totalPositionCollateral` from the contract, instead of recalculating here.